core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -190,6 +190,7 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
       .orNull
     numExecutors = Option(numExecutors)
       .getOrElse(sparkProperties.get("spark.executor.instances").orNull)
+    queue = Option(queue).orElse(sparkProperties.get("spark.yarn.queue")).getOrElse("default")
```

Comment:
ok, I'm keep it that way for consistency, just saying that the default queue name is not necessarily "default" either.